cmd/aperturectl/cmd/utils/controller.go

```diff
@@ -92,15 +107,38 @@ func (c *ControllerConn) InitFlags(flags *flag.FlagSet) {
 		"",
 		"FluxNinja ARC API Key to be used when using Cloud Controller",
 	)
+	flags.StringVar(
+		&c.config,
+		"config",
+		"",
+		"Path to the Aperture config file",
+	)
 }
 
 // PreRunE verifies flags (optionally loading kubeconfig) and should be run at PreRunE stage.
 func (c *ControllerConn) PreRunE(_ *cobra.Command, _ []string) error {
-	if c.controllerAddr == "" && !c.kube {
-		log.Info().Msg("Neither --controller nor --kube flag is set. Assuming --kube=true.")
+	// Fetching config from environment variable
+	if c.config == "" {
+		c.config = os.Getenv(configEnv)
+	}
+
+	// Fetching config from default location
+	if c.config == "" {
+		homeDir, err := os.UserHomeDir()
+		if err == nil {
+			c.config = filepath.Join(homeDir, ".aperturectl", "config")
+		}
+	}
+
+	if c.controllerAddr == "" && !c.kube && c.config == "" {
+		log.Info().Msg("Neither --controller nor --kube nor --config flag is set. Assuming --kube=true.")
 		c.kube = true
 	}
 
+	if c.config != "" && c.apiKey != "" {
+		return errors.New("--api-key cannot be used with --config")
+	}
```

Comment:
This check should be moved up, because it's ok if we use --api-key to override default config or config from env. (Actually, I think it's also ok if we have `--config xxx --api-key yyy and we override api key, but perhaps let's error, as that's weird)

Reply:
Made the changes as suggested